src/arch/arm/kernel/boot.c

```diff
@@ -304,7 +312,19 @@ BOOT_CODE static void release_secondary_cpus(void)
     /* Wait until all the secondary cores are done initialising */
     while (ksNumCPUs != CONFIG_MAX_NUM_NODES) {
         /* perform a memory release+acquire to get new values of ksNumCPUs */
-        __atomic_signal_fence(__ATOMIC_ACQ_REL);
+        __atomic_thread_fence(__ATOMIC_ACQ_REL);
```

Comment:
Can you make the change `__atomic_signal_fence`-> `__atomic_thread_fence` in a separate commit, because it fixes a potential `ksNumCPUs` update problem (ensure barriers instruction are created), which is unrelated to the clock sync test.

Reply:
Seems a `__ATOMIC_ACQUIRE` is sufficient then also, as we just have to ensure that every load is executed

Reply:
Normally I would agree that it should have been a separate commit. I considered that, but I deemed the problem to be too small to warrant it's own commit.

Because `ksNumCPUs` is a global variable there won't be any chance of optimising the access away, and the only difference is a missing barrier. The barrier is only necessary if there is data ordering, but there isn't any other data involved to order against until I introduced `clock_sync_test`.

Reply:
> Seems a __ATOMIC_ACQUIRE is sufficient then also

Probably, but that change does not warrant the slight chance of introducing any future bugs caused by other code changes IMHO.

I think the code should be rewritten to C11 standard atomics anyway, if you want to really improve it. `ksNumCPUs` is being used as a sort of lock, if the store and loads were done with acquire and release, then the `__atomic_thread_fence` after setting `clock_sync_test` could disappear altogether.